src/main/scala/org/broadinstitute/hail/stats/SparseGtBuilder.scala

```diff
@@ -0,0 +1,66 @@
+package org.broadinstitute.hail.stats
+
+import breeze.linalg.SparseVector
+import org.broadinstitute.hail.variant.Genotype
+
+import scala.collection.mutable
+
+// constructs SparseVector of genotype calls with missing values mean-imputed
+// if all genotypes are missing then all elements are NaN
+class SparseGtBuilder extends Serializable {
```

Comment:
Good call. It didn't start off this similar, but evolved in that direction so I'll take another look.